rust/lance-table/src/format/manifest.rs

```diff
@@ -214,9 +214,19 @@ impl Manifest {
     /// Replaces the metadata of the field with the given id with the given key-value pairs.
     ///
     /// If the field does not exist in the schema, this is a no-op.
-    pub fn replace_field_metadata(&mut self, field_id: i32, new_metadata: HashMap<String, String>) {
+    pub fn replace_field_metadata(
+        &mut self,
+        field_id: i32,
+        new_metadata: HashMap<String, String>,
+    ) -> Result<()> {
         if let Some(field) = self.schema.field_by_id_mut(field_id) {
             field.metadata = new_metadata;
+            Ok(())
+        } else {
+            Err(Error::FieldNotExists {
```

Comment:
sorry for the back and forth, I did not describe clearly. I think it is not worth creating a dedicated error for this, this is an invalid input of the replace_field_metadata function, so the original choice was correct. But we just need to have a more clear message like the new one you have - "Field does not exists: {field_id}"

Reply:
my bad